bcrypt.js

```diff
@@ -152,15 +152,15 @@ module.exports.compareSync = function compareSync(data, hash) {
 /// @param {String} hash expected hash
 /// @param {Function} cb callback(err, matched) - matched is true if hashed data matches hash
 module.exports.compare = function compare(data, hash, cb) {
-    if (data == null || hash == null) {
+    if (typeof data === 'function') {
         return process.nextTick(function() {
-            cb(new Error('data and hash arguments required'));
+            data(new Error('data and hash arguments required'));
```

Comment:
I tried to make it identical to `hash`, where it is assumed that if the user passes a function in place of parameters, the data and salt arguments are missing and the passed function is the callback which should be called with the error.

May be the error message could be rewritten as `'data and hash must be strings'` ?

If the user passes null for either the data or hash, and does not specify a callback, the Promise is rejected.

The previous (v0.8.7) behavior was to fail with `TypeError: cb is not a function`, But `hash` used to call the passed method and silently fail if no callback was specified.